### PR TITLE
fix not apparent which dropdown items can be activated

### PR DESCRIPTION
**Pull request by 0xF013**
[`* `#15507` not apparent which dropdown items can be activated`](http://community.openproject.org/work_packages/15507)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://0xf013-feature-15507-not-apparent-which-dropdown-items-can-be-a.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: karma/tests/directives/components/inaccessible-by-tab-directive-test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      expect(element.attr('tabindex')).to.eq("-1");
+    });
+
+    it('should not be -1 for false', function() {
+      scope.boolValue = false;
+      compile();
+      expect(typeof element.attr('tabindex')).to.eq("undefined");
+    });
+  });
+});
```
Comment (hschink) [inline karma/tests/directives/components/inaccessible-by-tab-directive-test.js:43]: Newline

File: app/assets/javascripts/angular/directives/components/inaccessible-by-tab-directive.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  return {
+    restrict: 'A',
+    link: function(scope, element, attrs) {
+      scope.oldTabIndex = element.attr("tabindex");
+      scope.$watch(
+        function() {
+          return scope.$eval(attrs.inaccessibleByTab);
+        },
+        function(isInaccessible) {
+          if( isInaccessible) {
```
Comment (hschink) [inline app/assets/javascripts/angular/directives/components/inaccessible-by-tab-directive.js]: Superfluous whitespace

File: karma/tests/directives/components/inaccessible-by-tab-directive-test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      scope.boolValue = true;
+      compile();
+      expect(element.attr('tabindex')).to.eq("-1");
+    });
+
+    it('should not be -1 for false', function() {
+      scope.boolValue = false;
+      compile();
+      expect(typeof element.attr('tabindex')).to.eq("undefined");
+    });
```
Comment (hschink) [inline karma/tests/directives/components/inaccessible-by-tab-directive-test.js:41]: I'd like to see an additional test for the case when the watch expression changes.

Comment (0xF013) [inline karma/tests/directives/components/inaccessible-by-tab-directive-test.js:41]: @hschink added

Comment (myabc) [inline karma/tests/directives/components/inaccessible-by-tab-directive-test.js:41]: @0xF013 you can also write `expect(element.attr('tabindex')).to.be.undefined`

**Comment by hschink** [comment]
@0xF013 karma tests fail

**Comment by hschink** [comment]
@0xF013 Tests fail

File: app/assets/javascripts/angular/ui_components/inaccessible-by-tab-directive.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  return {
+    restrict: 'A',
+    scope: {
+      isInaccessible: "=inaccessibleByTab"
+    },
+    link: function(scope, element, attrs) {
+      scope.oldTabIndex = element.attr("tabindex");
+      scope.$watch('isInaccessible',
+        function(isInaccessible) {
+          if( isInaccessible) {
```
Comment (hschink) [inline app/assets/javascripts/angular/ui_components/inaccessible-by-tab-directive.js]: Missing and superfluous space `if (isIn...`

File: karma/tests/directives/components/inaccessible-by-tab-directive-test.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    it('should change if the directive attr changes', function() {
+      scope.boolValue = false;
+      compile();
+      expect(element.attr('tabindex')).to.be.undefined;
+      scope.boolValue = true;
+      scope.$digest();
+      expect(element.attr('tabindex')).to.eq("-1");
+    });
+  });
+});
```
Comment (hschink) [inline karma/tests/directives/components/inaccessible-by-tab-directive-test.js:43]: Newline

**Comment by hschink** [comment]
As soon as the minor issues above are solved this can be merged. Good job @0xF013! :+1:

**Comment by 0xF013** [comment]
@hschink amended.

**Pull request merged**